src/ECL/ECL.hs

```diff
@@ -26,11 +30,26 @@ type Binders lit = [Binder lit]
 data Binder lit
   = Var (Maybe Name)
   | Lit lit
-  | Tagged Name (Binder lit)
+  | Tagged Name [Binder lit]
```

Comment:
Do you really need a list here? You could encode this using `Product`. If you do want it, I think you mean`Binders`?